src/numbers.jl

```diff
@@ -52,3 +52,11 @@ end
 real_dimension(::RealNumbers) = 1
 real_dimension(::ComplexNumbers) = 2
 real_dimension(::QuaternionNumbers) = 4
+
+@doc raw"""
+    number_system(M::Manifold)
+
+return the number system a manifold is based on. The default of the number system is
+real-valued, i.e. `number_system(M) = ℝ`.
+"""
+number_system(M::Manifold) = ℝ
```

Comment:
I'm not sure if this default is a good idea, you've already made an error due to this (`DefaultManifold` can be a complex manifold, depending on its number system argument).

Reply:
Thanks for noticing, that I missed implementing it for `DefaultManifold`. The idea for the default is, that manifolds, that do not have such a field are (implicitly) assumed to be real-valued. Though looking for an example I think this is only applicable for the `Torus`. I would like to make it easy for a user to have this by a default, since I think this function might be used quite often in the end.

I will add the function for `DefaultManifold` – but what do you think about the default? Is it useful or too dangerous?

Reply:
OK, let's keep the default but also let's be careful about testing this method for each manifold.

By the way, what should one do if they want a real-coefficient basis of a product of a real and a complex manifold?

Reply:
That depends on what coefficients you mean? The basis vectors? Then this should be an error since the second basis is complex.

But does the Product Basis have a field? It does not need one I think? It should be inferred automatically.

Reply:
I mean real coefficients as in coefficients of linear combination of basis vectors. Could we assume that for a real Euclidean space the complex basis is the same as the real basis?

Currently product bases assume that all multiplied bases are over the same number system.

Reply:
Well it is not that reasonable to have real euclidean with complex base, since it might yield contingent points as soon as you produce non real entries.

But despite that I think it is ok to assume that especially in a product manifold. So in that sense we can keep the current assumption (and not be too restrictive).